toonz/sources/tnztools/cursormanager.cpp

```diff
@@ -132,12 +146,75 @@ class CursorManager {  // singleton
   const CursorData &getCursorData(int cursorType) {
     // se e' gia' in tabella lo restituisco
     std::map<int, CursorData>::iterator it;
-    it = m_cursors.find(cursorType);
-    if (it != m_cursors.end()) return it->second;
+
+    if (Preferences::instance()->getCursorBrushStyle() == "Simple")
+      cursorType = ToolCursor::PenCursor;
+
+    if (cursorType == ToolCursor::PenCursor) {
+      QString brushType = Preferences::instance()->getCursorBrushType();
+      if (brushType == "Large")
+        cursorType = ToolCursor::PenLargeCursor;
+      else if (brushType == "Crosshair")
+        cursorType = ToolCursor::PenCrosshairCursor;
+    }
+
+    bool useLeft =
+        (Preferences::instance()->getCursorBrushStyle() == "Left-Handed");
+    if (useLeft) {
+      it = m_cursorsLeft.find(cursorType);
+      if (it != m_cursorsLeft.end()) return it->second;
+    } else {
+      it = m_cursors.find(cursorType);
+      if (it != m_cursors.end()) return it->second;
+    }
 
     int decorationsFlag = cursorType & ~(0xFF);
     int baseCursorType  = cursorType & 0xFF;
 
+    switch (baseCursorType) {
+    case ToolCursor::FxGadgetCursor:
+      decorationsFlag |= ToolCursor::Ex_FX;
+      break;
```

Comment:
Here you are trying to find specific tool type to add a decoration which cannot be generated by simple mirroring.
Instead, how about including such decoration information in the cursor flag like as follows?

```
-------------------- toonz/sources/include/tools/cursors.h --------------------
@@ -75,7 +75,7 @@ enum {
   MoveZCursor,
   ScaleGlobalCursor,
   ScaleHVCursor,
-  FxGadgetCursor,
+  FxGadgetCursorBase,
   RulerModifyCursor,
   RulerNewCursor,
 
@@ -92,7 +92,9 @@ enum {
   Ex_StyleLine        = 0x20000,
   Ex_StyleArea        = 0x40000,
   Ex_RGB              = 0x80000,
-  Ex_HV               = 0x100000
+  Ex_HV               = 0x100000,
+
+  FxGadgetCursor = FxGadgetCursorBase | Ex_FX
 };
 
 }  // namespace

------------------- toonz/sources/tnztools/cursormanager.cpp -------------------
@@ -34,7 +34,7 @@ const struct {
     {ToolCursor::MoveNSCursor, "move_ns", 15, 15, false},
     {ToolCursor::DisableCursor, "disable", 15, 15, false},
     {ToolCursor::MoveZCursor, "move_z_notext", 15, 15, false},
-    {ToolCursor::FxGadgetCursor, "edit_FX_notext", 11, 6, true},
+    {ToolCursor::FxGadgetCursorBase, "edit_FX_notext", 11, 6, true},
     {ToolCursor::FlipHCursor, "flip_h", 15, 15, false},
     {ToolCursor::FlipVCursor, "flip_v", 15, 15, false},
     {ToolCursor::IronCursor, "iron", 15, 15, true},
```

Reply:
Changes made. Thanks for the recommendation